Source/inv.cpp

```diff
@@ -207,21 +207,19 @@ void DrawInv(const CelOutputBuffer &out)
 		if (!myPlayer.InvBody[slot].isEmpty()) {
 			int screenX = slotPos[slot].x;
 			int screenY = slotPos[slot].y;
-			InvDrawSlotBack(out, RIGHT_PANEL_X + screenX, screenY, slotSize[slot].Width * INV_SLOT_SIZE_PX, slotSize[slot].Height * INV_SLOT_SIZE_PX);
+			InvDrawSlotBack(out, { RIGHT_PANEL_X + screenX, screenY }, { slotSize[slot].Width * InventorySlotSizeInPixels.Width, slotSize[slot].Height * InventorySlotSizeInPixels.Height });
 
 			int frame = myPlayer.InvBody[slot]._iCurs + CURSOR_FIRSTITEM;
 
-			int frameW;
-			int frameH;
-			std::tie(frameW, frameH) = GetInvItemSize(frame);
+			auto frameSize = GetInvItemSize(frame);
 
 			// calc item offsets for weapons smaller than 2x3 slots
 			if (slot == INVLOC_HAND_LEFT) {
-				screenX += frameW == INV_SLOT_SIZE_PX ? 14 : 0;
-				screenY += frameH == (3 * INV_SLOT_SIZE_PX) ? 0 : -14;
+				screenX += frameSize.Width == InventorySlotSizeInPixels.Width ? 14 : 0;
+				screenY += frameSize.Height == (3 * InventorySlotSizeInPixels.Height) ? 0 : -14;
 			} else if (slot == INVLOC_HAND_RIGHT) {
-				screenX += frameW == INV_SLOT_SIZE_PX ? 13 : 1;
-				screenY += frameH == 3 * INV_SLOT_SIZE_PX ? 0 : -14;
+				screenX += frameSize.Width == InventorySlotSizeInPixels.Width ? 13 : 1;
+				screenY += frameSize.Height == 3 * InventorySlotSizeInPixels.Height ? 0 : -14;

```

Comment:
```suggestion
				screenY += frameSize.Height == (3 * InventorySlotSizeInPixels.Height) ? 0 : -14;

```

Reply:
Why was this not raising a warning in the old code?

Reply:
It's probably not considered ambitious by GCC. Maybe VS would have complained if we ran it at a higher level of pedanticism.